src/rust/engine/process_execution/src/immutable_inputs.rs

```diff
@@ -94,58 +82,21 @@ impl ImmutableInputs {
             digest, e
           )
         })?;
+
+        let dest = chroot.path().join(digest.hash.to_hex());
```

Comment:
I'm totally lost. How is this reproducible digest suffix any more anonymous than the tmpdir its being joined with?

Reply:
The creation of the tempdir is guaranteed not to collide (via OS facilities, presumably), so each materialization attempt is in a fresh temporary directory. The digest suffix is a convenience for readability: it's the temp dir that is avoiding collisions.

Reply:
On the LHS:
```rust
self.store..materialize_directory(
  chroot.path().to_path_buf(),
  directory_digest,
  Permissions::ReadOnly,
)
```

On the RHS - paraphrasing with an inline:
```rust
self.store..materialize_directory(
  chroot.path().join(digest.hash.to_hex()),
  directory_digest,
  Permissions::ReadOnly,
)
```

So the only change is to use the tempdir as-is instead of renaming it to a fixed `self.workdir.path().join(digest.hash.to_hex())` dest - Got it.

Why was the self.workdir rooting important before and not now? Or was it never?

Reply:
Ok, so self.workdir is already a TempDir. So we create a TempDir in a TempDir, as we always did, but no longer rename from `<tmpdir>/<tmpdir>` -> `<tmpdir>/<digest>` we now just directly create `<tmpdir>/<tmpdir>/<digest>`. It would seem the outer <tmpdir> could just be a normal dir now, but that's only mildly confusing as is.